src/helpers/graph.js

```diff
@@ -0,0 +1,42 @@
+export function removeNoteFromGraph(nodes, edges, noteToDelete) {
+  let newNodes = [ ...nodes ]
+  let newEdges = [ ...edges ]
+  let parentEdge = newEdges.find(edge => edge.to === noteToDelete.id)
+
+  removeChild(noteToDelete.id)
+  newNodes = newEdges.find(edge => edge.from === parentEdge.from) ? //check if there are siblings of deleted node if not revoke parant status
+             newNodes : newNodes.map(node => { return node.id === parentEdge.from ? { ...node, group: 'children' } : node })
+  return { notes: newNodes, edges: newEdges }
+
+  function removeChild(childNoteId) {
+    newNodes = newNodes.filter(note => note.id !== childNoteId)
+    if(newEdges.find(edge => edge.from === childNoteId)) { // check if the node has children
+      newEdges.forEach(edge => {
+        if (edge.from === childNoteId) {
+          removeChild(edge.to)
+          newEdges = newEdges.filter(e => (e.from !== childNoteId))
+        }
+      })
+      newEdges = newEdges.filter(e => e.to !== childNoteId) // remove the edge from parent to the deleted node
+    } else {
+      newEdges = newEdges.filter(e => e.to !== childNoteId)
+      return;
+    }
+  }
+}
+
+export function updateGroupOfOldParent(nodes, edges, newParentId, oldParentId) {
+  nodes = nodes.map(node => {
+    if (node.id === newParentId) return { ...node, group: 'parent' }
+    else if (node.id === oldParentId && edges.find(edge => edge.from === node.id) === undefined){
+      return { ...node, group: 'children' }
+    } 
+    return node
+  })
+  return nodes

```

Comment:
I think what you're doing in this function could be done easier and in less confusing way:
- if a node has children, it's group should be `parent`.
- otherwise, it's group should be `children`.

So instead of having a weird method named `updateGroupOfOldParent`, you could have a method 
```
const updateNodeGroup = node => hasChildren(node) ? {...node, group: 'parent'} : {...node, group: 'children'};
```

And then whenever you need to update any node's group, be it an old parent, or a new parent, you can just do
```
const oldParentWithUpdatedGroup = updateNodeGroup(oldParent);
```